content/blog/2021/10/new-architecture-to-ease-kubewarden-administrators-lives.md

```diff
@@ -0,0 +1,268 @@
+---
+title: A new architecture to ease Kubewarden administrators' lives
+authors:
+    - Víctor Cuadrado Juan
+date: 2021-10-01
+---
+
+We are pleased to announce a new architecture for the Kubewarden stack, in line
+with its journey to maturity:
+
+The introduction of a **PolicyServer** Custom Resource Definition (CRD) which
+allows users to describe a policy-server Deployment, together with binding
+**ClusterAdmissionPolicies** to a specific **PolicyServer** instance.
+
+These 2 changes are accompanied by a multitude of improvements to make Kubewarden
+more comfortable for Kubernetes Administrators, such as validation for
+Kuberwarden Custom Resources, improvements in Helm Charts, Status and
+Conditions for **ClusterAdmissionPolicies**.
+
+For a comprehensive list, see the [release notes](TODO) of our latest Helm
+charts.
+
+## How does it look like?
+
+In previous versions, the Kubewarden Controller instantiated a single Deployment
+of policy-server. That policy-server was configured via a ConfigMap, which
+contained the Deployment options (image, replicas, ...), and a list of policies
+to be loaded, with information on where to pull them from, their configuration
+options and so on.
+
+With the addition of the new **PolicyServer** Custom Resource, administrators
+have a better UX, since they can define as many policy servers as they need, and
+get to select what PolicyServer each ClusterAdmissionPolicy targets. Let's see a
+diagram of the new architecture:
+
+{{< figure class="center" src="/images/how-it-works-kubewarden.svg" width="90%" alt="new architecture diagram">}}
+
+On the diagram, notice the 2 separate PolicyServer Deployments in cyan and mauve
+(right), created as specified in the 2 PolicyServer resources (left).
+
+Each policy server loads different policies -- all ClusterAdmissionPolicies that
+target that specific policy server. The new PolicyServer Custom Resource is
+cluster-wide, which means that is identifiable by its unique `name`. Here is an
+example of a PolicyServer named `tenant-a`:
+
+```yaml
+---
+apiVersion: policies.kubewarden.io/v1alpha2
+kind: PolicyServer
+metadata:
+  name: tenant-a
+spec:
+  image: ghcr.io/kubewarden/policy-server:v0.1.10
+  replicas: 1
+```
+
+The PolicyServer Custom Resource also accepts an optional
+`spec.serviceAccountName` to be associated with (if not set, as here, the
+Namespace ServiceAccount will be used).
```

Comment:
```suggestion
Namespace default ServiceAccount will be used).
```

Reply:
Fixed in https://github.com/kubewarden/kubewarden.io/pull/33/commits/48ba5b1913d909022e7e42df9b1894fc5166a813.